### ISSUE TITLE
No longer works in MacOS High Sierra

### ISSUE THREAD

**Issue by Raptor399** [Swordfish90/cool-retro-term#366]
After the update to MacOS High Sierra, the shell of cool-retro-term never starts up. The visual effects of the terminal (flicker, scanline, etc) are there, but the cursor remains in the top left of the screen:

![image](https://user-images.githubusercontent.com/879970/30836411-164cd74a-a22c-11e7-9ebf-86231bcc98e6.png)

As a result, cool-retro-term no longer works.

**Comment by miho** [comment]
Noticed the same.

**Comment by Shunajlov** [comment]
+1. Please, fix it, i love this styled terminal

**Comment by bnaught0n** [comment]
Here's what I get when I start from cli:
----
System:MacOS user$ ./cool-retro-term 
Using an unsupported variable-width font in the terminal.  This may produce display errors.
Can't open a pseudo teletype
default KB_LAYOUT_DIR:  "/Applications/cool-retro-term.app/Contents/MacOS/../PlugIns/QMLTermWidget/kb-layouts"
QObject::connect: No such slot Konsole::TerminalDisplay_QML_61::close()
loadAllColorSchemes
Session::run() - program: "/bin/bash"
Session::run() - arguments: ("-i", "-l")
Unable to set terminal attributes.
started!
Using an unsupported variable-width font in the terminal.  This may produce display errors.
System:~ user$ Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

**Comment by wyne** [comment]
In addition to the above, any input results in this log:
```
QIODevice::write (KPtyDevice): device not open
QIODevice::write (KPtyDevice): device not open
```

**Comment by donatj** [comment]
I am also seeing the following in Console. Not sure if this helps

```
default	23:48:35.379386 -0500	cool-retro-term	LSExceptions shared instance invalidated for timeout.
default	23:49:16.420655 -0500	cool-retro-term	OSErr AERemoveEventHandler(AEEventClass, AEEventID, AEEventHandlerUPP, Boolean)(aevt,quit handler=0x7fff3e9b09fc isSys=NO) err=0/noErr
default	23:49:16.420686 -0500	cool-retro-term	OSErr AERemoveEventHandler(AEEventClass, AEEventID, AEEventHandlerUPP, Boolean)(GURL,GURL handler=0x7fff3e9b09fc isSys=NO) err=0/noErr
```

**Comment by extigy** [comment]
I got this mostly working, at least for me. Begin by following the normal macOS instructions, reproduced here:
```
brew install qt5
git clone --recursive https://github.com/Swordfish90/cool-retro-term.git
export CPPFLAGS="-I/usr/local/opt/qt5/include"
export LDFLAGS="-L/usr/local/opt/qt5/lib"
export PATH=/usr/local/opt/qt5/bin:$PATH
cd cool-retro-term
```
At this point, edit the file `qmltermwidget/qmltermwidget.pro` and change line 11 to read:
`macx:DEFINES += HAVE_UTMPX _UTMPX_COMPAT HAVE_PTSNAME HAVE_UNLOCKPT HAVE_GRANTPT`

Then continue with the rest:
```
qmake && make
mkdir cool-retro-term.app/Contents/PlugIns
cp -r qmltermwidget/QMLTermWidget cool-retro-term.app/Contents/PlugIns
open cool-retro-term.app
```

Unfortunately, when building from source I see `Illegal Instruction: 4` on closing the app, but that was happening even before changing `qmltermwidget.pro`. Perhaps I'll try compiling with an older version of Xcode selected to see if that helps.

**Comment by Raptor399** [comment]
Fix confirmed, pull request created.

**Comment by Swordfish90** [comment]
Hi guys. Thank you for the quick fix. Sadly I'm out of apple machines at the moment, can you confirm this is working and not introducing regressions (such as on older versions of the OS)?

**Comment by giann** [comment]
@Swordfish90 I can confirm it works on High Sierra. I'll try tonight on Sierra.

**Comment by Shunajlov** [comment]
@giann Can you please share an .app file built for High Sierra?

**Comment by extigy** [comment]
Glad the change works for others. I wasn't sure how older versions of macOS handle it and don't have an old machine to test, so didn't pull request.

However, a quick web search brings up [this file](http://ab-initio.mit.edu/octave-Faddeeva/gnulib/lib/posix_openpt.c) which suggests in a comment that the posix_openpt functions are available from OSX >= 10.4, so hopefully it just works!

**Comment by Raptor399** [comment]
It took me half a day to revive an old Macbook Pro with OSX Yosemite (10.10.4) and get the correct version of Xcode, Qt and everything on there...

The result is that I can confirm cool-retro-term with the fix still works on ancient Macs running unsupported operating systems! ;-)

**Comment by Swordfish90** [comment]
Ok guys. I just closed this and forced a friend to build a dmg file. Version 1.0.1 should be working. Thank you @extigy and @Raptor399 on this one.

**Comment by despair86** [comment]
Interesting. This also fixed Solaris 2.11
